### PR TITLE
#110 - Display Context - Rules definition

### PR DESCRIPTION
**Pull request by jplucinski**
Issue #110 specify rule for implicit display context, but after research I found few more rules that defines when developer should use it implicitly or explicitly. This PR contains all of them. For more information please refer to issue comments.

### CODE REVIEW

File: src/main/java/com/cognifide/aemrules/htl/checks/EventMandatoryDisplayContextCheck.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    private static final Pattern LITERAL_EXPRESION_PATTERN = Pattern.compile("\\$\\{.*}");
+
+    private static final String DISPLAY_CONTEXT_ATTRIBUTE = "context";
+
+    @Override
+    public void startElement(TagNode node) {
+        node.getAttributes()
+                .stream()
+                .filter(attribute -> StringUtils.startsWith(attribute.getName(), "on"))
+                .filter(attribute -> LITERAL_EXPRESION_PATTERN.matcher(attribute.getValue()).find() && !attribute.getValue().contains(DISPLAY_CONTEXT_ATTRIBUTE))
```
Comment (piotr-wilczynski) [inline src/main/java/com/cognifide/aemrules/htl/checks/EventMandatoryDisplayContextCheck.java]: please extract it to method, for better readability

File: src/main/java/com/cognifide/aemrules/htl/checks/InlineStyleMandatoryDisplayContextCheck.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    private static final Pattern LITERAL_EXPRESION_PATTERN = Pattern.compile("\\$\\{.*}");
+
+    private static final String DISPLAY_CONTEXT_ATTRIBUTE = "context";
+
+    @Override
+    public void startElement(TagNode node) {
+        node.getAttributes()
+                .stream()
+                .filter(attribute -> StringUtils.equals(attribute.getName(), "style"))
+                .filter(attribute -> LITERAL_EXPRESION_PATTERN.matcher(attribute.getValue()).find() && !attribute.getValue().contains(DISPLAY_CONTEXT_ATTRIBUTE))
```
Comment (piotr-wilczynski) [inline src/main/java/com/cognifide/aemrules/htl/checks/InlineStyleMandatoryDisplayContextCheck.java]: same here

File: src/main/resources/rules/HTL-7.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+For style and script tags display context definition is mandatory.
+
+== Noncompliant Code Example
```
Comment (piotr-wilczynski) [inline src/main/resources/rules/HTL-7.md]: please be consistent in Noncompliant & Compliant order

File: README.md [hunk 4]
```diff
@@ -55,6 +55,15 @@ gradlew sonarQube -DsonarRunner.aemVersion=6.4
 
 Below you will find descriptions of all rules available in **AEM Rules for SonarQube** plugin.
 
+- **HTL-6** HTL automatically recognises the context for HTML output
```
Comment (GulderBone) [inline README.md]: Could we merge this branch: https://github.com/Cognifide/AEM-Rules-for-SonarQube/pull/147/files#diff-04c6e90faac2675aa89e2176d2eec7d8
and move your descriptions to keep that convention.

**Comment by GulderBone** [comment]
You should update the number of rules in `HtlProfileTest` class.

**Comment by jplucinski** [comment]
> You should update the number of rules in `HtlProfileTest` class.

Let's discuss about the meaning of a given class. Current implementation looks like unfinished.

**Pull request merged**